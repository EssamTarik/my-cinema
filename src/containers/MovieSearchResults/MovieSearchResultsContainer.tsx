import React from "react";
import MovieGridContainer from "../MovieGrid";
import { IProps } from "./interfaces";

const MovieSearchResultsContainer = ({ movieSearchState }: IProps) => {
  const { isFetching, isFetched, data, error } = movieSearchState;
  if (isFetching || !isFetched || !data) {
    return null;
  }
  if (error) {
    return <div className="movie-search__error">{error}</div>;
  }

  const { results: movies } = data;
  return <MovieGridContainer movies={movies} />;
};

export default MovieSearchResultsContainer;
