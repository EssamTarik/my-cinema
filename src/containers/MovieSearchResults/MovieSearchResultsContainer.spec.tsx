import React from "react";
import { render, cleanup } from "@testing-library/react";

import MovieSearchResultsContainer from "./MovieSearchResultsContainer";

jest.mock("../MovieGrid", () => () => (
  <div className="movie-grid">movie grid</div>
));

const movieSearchResultsMock = jest.requireMock("movieSearchResults");

describe("MovieSearchResultsContainer", () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <MovieSearchResultsContainer
        movieSearchState={{
          isFetched: true,
          isFetching: false,
          data: movieSearchResultsMock,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("returns null if no results", () => {
    let { container } = render(
      <MovieSearchResultsContainer
        movieSearchState={{ isFetched: false, isFetching: true }}
      />
    );
    let resultGrid = container.querySelector(".movie-grid");
    expect(resultGrid).toEqual(null);

    container = render(
      <MovieSearchResultsContainer
        movieSearchState={{
          isFetched: true,
          isFetching: false,
          data: undefined,
        }}
      />
    ).container;

    resultGrid = container.querySelector(".movie-grid");
    expect(resultGrid).toEqual(null);

    container = render(
      <MovieSearchResultsContainer
        movieSearchState={{
          isFetched: true,
          isFetching: false,
          data: movieSearchResultsMock,
          error: "error",
        }}
      />
    ).container;

    resultGrid = container.querySelector(".movie-grid");
    expect(resultGrid).toEqual(null);
  });

  it("returns movie grid with valid results", () => {
    const { container } = render(
      <MovieSearchResultsContainer
        movieSearchState={{
          isFetched: true,
          isFetching: false,
          data: movieSearchResultsMock,
        }}
      />
    );

    const resultGrid = container.querySelector(".movie-grid");
    expect(resultGrid).not.toEqual(null);
  });

  it("returns error in case of error", () => {
    const errorMessage = "Failed to fetch";
    const { container } = render(
      <MovieSearchResultsContainer
        movieSearchState={{
          isFetched: true,
          isFetching: false,
          error: errorMessage,
          data: movieSearchResultsMock,
        }}
      />
    );

    const errorDiv = container.querySelector(".movie-search__error");
    expect(errorDiv).not.toEqual(null);
  });
});
