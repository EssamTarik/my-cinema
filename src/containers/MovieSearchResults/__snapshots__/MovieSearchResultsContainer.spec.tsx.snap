// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchResultsContainer renders 1`] = `
<DocumentFragment>
  <div
    class="movie-grid"
  >
    movie grid
  </div>
</DocumentFragment>
`;
